src/compiler/checker.ts

```diff
@@ -60,6 +72,35 @@ namespace ts {
         undefinedSymbol.declarations = [];
         const argumentsSymbol = createSymbol(SymbolFlags.Property | SymbolFlags.Transient, "arguments");
 
+        const {typeToString, getSymbolDisplayBuilder, symbolToString, signatureToString, typePredicateToString, visibilityToString} =
+            createWriter(
+                createSymbol,
+                getSymbolOfNode,
+                compilerOptions,
+                languageVersion,
+                getTypeParametersOfClassOrInterface,
+                getAccessibleSymbolChain,
+                needsQualification,
+                getQualifiedLeftMeaning,
+                hasExternalModuleSymbol,
+                isTypeAny,
+                () => nothingType,

```

Comment:
You mean, basically surround the whole thing with `{}` and add `getNothingType: ` in front of `() => nothingType`? That seems reasonable, especially if these collections get less ad-hoc and might be re-used between other 'modules'.